app/views/questionnaire.py

```diff
@@ -497,40 +499,52 @@ def _redirect_to_location(collection_id, eq_id, form_type, location):
                             group_instance=location.group_instance, block_id=location.block_id))
 
 
-def _get_context(full_routing_path, block, current_location, answer_store):
+def _get_context(full_routing_path, block, current_location, post_form):
+    metadata = get_metadata(current_user)
+    answer_store = get_answer_store(current_user)
+
+    schema_context = _get_schema_context(full_routing_path, current_location, metadata, answer_store)
+
+    block = renderer.render(block, **schema_context)
+
+    variables = g.schema_json.get('variables')
+    if variables:
+        variables = renderer.render(variables, **schema_context)
+
+    if post_form:
+        return {'form': post_form, 'block': block, 'variables': variables}
 
     error_messages = SchemaHelper.get_messages(g.schema_json)
     form, template_params = get_form_for_location(block, current_location, answer_store, error_messages)
 
-    content = {'form': form, 'block': block}
+    context = {'form': form, 'block': block}
     if template_params:
-        content.update(template_params)
+        context.update(template_params)
 
     if block['type'] == 'Summary':
-        metadata = get_metadata(current_user)
-        aliases = SchemaHelper.get_aliases(g.schema_json)
-        schema_context = build_schema_context(metadata=metadata,
-                                              aliases=aliases,
-                                              answer_store=answer_store,
-                                              routing_path=full_routing_path)
         rendered_schema_json = renderer.render(g.schema_json, **schema_context)
-        content.update({'summary': build_summary_rendering_context(rendered_schema_json, answer_store, metadata)})
+        context.update({'summary': build_summary_rendering_context(rendered_schema_json, answer_store, metadata)})
+
```

Comment:
I think the summary block needs to be dealt with first, as it doesn't need `block` and `form` but they're both generated anyway.

Reply:
It appears summary does require "form". Have pushed the rest of the changes and will investigate this further.